src/page/ManageEvent/ModifyEvent/index.tsx

```diff
@@ -0,0 +1,335 @@
+  };
+
+  const [eventInfo, setEventInfo] = useState<EventInfo>(initialState);
+  const [error, setError] = useState(initialError);
+  const [editor, setEditor] = useState(event.content);
+
+  const { mutate: modifyEvent, isPending } = useModifyEvent(event.shop_id, event.event_id);
+  const navigate = useNavigate();
+
+  const changeTitle = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.value.length <= 25) setEventInfo({ ...eventInfo, title: e.target.value });
+    setError({ ...error, title: false });
+  };
+  const changeStartYear = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value.replaceAll(/[\D]/gi, '');
+    if (value.length <= 4) {
+      setEventInfo({ ...eventInfo, start_date: { ...eventInfo.start_date, year: value } });
+    }
+    setError({ ...error, date: false });
+  };
+  const changeStartMonth = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value.replaceAll(/[\D]/gi, '');
+    if (value.length <= 2) {
+      setEventInfo({ ...eventInfo, start_date: { ...eventInfo.start_date, month: value } });
+    }
+    setError({ ...error, date: false });
+  };
+  const changeStartDate = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value.replaceAll(/[\D]/gi, '');
+    if (value.length <= 2) {
+      setEventInfo({ ...eventInfo, start_date: { ...eventInfo.start_date, date: value } });
+    }
+    setError({ ...error, date: false });
+  };
+  const changeEndYear = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value.replaceAll(/[\D]/gi, '');
+    if (value.length <= 4) {
+      setEventInfo({ ...eventInfo, end_date: { ...eventInfo.end_date, year: value } });
+    }
+    setError({ ...error, date: false });
+  };
+  const changeEndMonth = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value.replaceAll(/[\D]/gi, '');
+    if (value.length <= 2) {
+      setEventInfo({ ...eventInfo, end_date: { ...eventInfo.end_date, month: value } });
+    }
+    setError({ ...error, date: false });
+  };
+  const changeEndDate = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value.replaceAll(/[\D]/gi, '');
+    if (value.length <= 2) {
+      setEventInfo({ ...eventInfo, end_date: { ...eventInfo.end_date, date: value } });
+    }
+    setError({ ...error, date: false });
+  };
+
+  const validation = () => {
+    let flag = false;
+    if (eventInfo.title.length === 0) {
+      setError((prevError) => ({ ...prevError, title: true }));
+      flag = true;
+    }
+    if (editor.length === 0) {
+      setError((prevError) => ({ ...prevError, content: true }));
+      flag = true;
+    }
+    if (eventInfo.start_date.year.length === 0
+      || eventInfo.start_date.month.length === 0
+      || eventInfo.start_date.date.length === 0
+      || eventInfo.end_date.year.length === 0
+      || eventInfo.end_date.month.length === 0
+      || eventInfo.end_date.date.length === 0
+    ) {
```

Comment:
뭔가 이것도 옆으로 빼면 참 좋을 것 같긴한데 뭔가 이런 날짜 검증과 관련되어서는 많이 사용될 것도 같네요
```suggestion
   const dateValidationCheck = (date: Date) => {
      return date.year.length === 0 || date.month.length === 0 || date.date.length === 0;
   }  
```
대충 이런식으로 공용함수로 빼놓고 나중에 재사용하면 좋을 것 같습니다.

Reply:
91ebd43f0c87ea40dc87e9c003b6728404e15849

마찬가지로 적용했습니다